### PR TITLE
feat(cli): properly shutdown subprocesses on abort

### LINKED ISSUE #771: Aborting a hanging cdktf deploy leaves the underlying Terraform CLI process alive
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### cdktf & Language Versions

cdktf 0.4.0
Terraform 0.14.9

### Debug Output

Process (\# 2) still alive..
Screenshot 2021-06-10 at 10 29 59

### Expected Behavior

It should also stop the spawned Terraform CLI process when stopping the cdktf-cli.

### Actual Behavior

The Terraform CLI was still running and thereby keeping a lock on my local state which I couldn't unlock.

### Steps to Reproduce

I had expired AWS credentials in my environment from the day prior which somehow caused the AWS Terraform provider to hang. While waiting for the command to finish, I noticed this and aborted the `cdktf deploy` via `ctrl + c`.  
Afterwards the Terraform CLI process was still running (see screenshot above) and was keeping me from doing a deploy / diff with new credentials because it kept a state lock.  

### References

* #0000

### PR DESCRIPTION
**Pull request by DanielMSchmidt**
This PR ensures that when we receive a termination signal we stop all corresponding node processes. I'd consider it out of scope to stop / delete terraform cloud runs. I'm not even sure if stopping requests for TFC is worth it.

Closes #771

You can see in the watch tab below how the terraform apply command is aborted
abort

Created https://github.com/hashicorp/terraform-cdk/issues/1607 as follow up to abort TFC runs

### CODE REVIEW

File: packages/cdktf-cli/bin/cdktf [hunk 1]
```diff
@@ -1,2 +1,2 @@
-#!/usr/bin/env node
-require('./cdktf.js');
\ No newline at end of file
+#!/usr/bin/env node --experimental-abortcontroller
```
Comment (ansgarm) [inline packages/cdktf-cli/bin/cdktf:1]: Maybe we can remove this on node v14.17?

File: packages/cdktf-cli/bin/cmds/helper/check-environment.ts [hunk 2]
```diff
@@ -45,7 +45,7 @@ async function checkGoVersion() {
 
 async function checkNodeVersion() {
   const out = await getBinaryVersion("node", "--version");
-  throwIfLowerVersion("Node.js", "14.0.0", out);
+  throwIfLowerVersion("Node.js", "14.17.0", out);
```
Comment (ansgarm) [inline packages/cdktf-cli/bin/cmds/helper/check-environment.ts:48]: We also need to update various docs, I think 😬

File: packages/cdktf-cli/lib/models/terraform-cloud.ts [hunk 3]
```diff
@@ -114,6 +114,8 @@ export class TerraformCloud implements Terraform {
   public run?: TerraformCloudClient.Run;
 
   constructor(
+    // TODO: teach TerraformCloudClient abort signals
```
Comment (ansgarm) [inline packages/cdktf-cli/lib/models/terraform-cloud.ts:117]: todo left, link to issue?

File: packages/cdktf-cli/lib/server/WatchClient.ts [hunk 4]
```diff
@@ -244,9 +250,12 @@ export class WatchClient {
 
   // todo: optimization: cache instance as long as backend does not change
   private async getTerraform(): Promise<Terraform> {
+    // TOOD: implement abort properly for watch
```
Comment (ansgarm) [inline packages/cdktf-cli/lib/server/WatchClient.ts:253]: todo left?

**Review by ansgarm — APPROVED** [review]

File: packages/cdktf-cli/bin/cdktf (hunk 1 above)
Comment (DanielMSchmidt) [reply packages/cdktf-cli/bin/cdktf:1]: https://nodejs.org/docs/latest-v14.x/api/globals.html#globals_class_abortcontroller seems to indicate that it's needed for 14.X

**Pull request merged**